maintainer/channels.py

```diff
@@ -0,0 +1,145 @@
+        '''
+        self.qtime = qtime
+        self.connection = 'http://193.70.3.178:9652/'
+        self.password = configuration.auth_pwd
+        self.username = configuration.auth_user
+
+    def _requests_get(self, endpoint):
+        return requests.get('{}{}'.format(
+            self.connection, endpoint), auth=(self.username, self.password)).json()
+
+    def _requests_delete(self, endpoint):
+        requests.delete('{}{}'.format(
+                self.connection, endpoint), auth=(self.username, self.password))
+
+    def _requests_post(self, endpoint, payload):
+        requests.post('{}{}'.format(self.connection, endpoint), auth=(
+            self.username, self.password), data=payload)
+
+    def channels_basic_info(self):
+        '''
+        Get basic information about every channel on server.
+        In details it tells us names of channels, how many users are on specific
+        channels, and channel_id with parent_id.
+        Output: channels_basic_info as list of dictionaries.
+        '''
+        channels = self._requests_get('channels')
+        channels_basic_info = []
+        for channel in channels:
+            channels_basic_info.append(channel)
+        return channels_basic_info
+
+    def channels_detailed_information(self):
+        '''
+        Method is returning more data than channels_basic_info method.
+        In this method we are receiving important data such as: channel_topic
+        or seconds_empty that allow us to see if there is any status written
+        into topic or if channel is used at all.
+        There is no input value from outside of class.
+        Output: channels_detailed_info as dictionary of dictionaries in form of:
+            cid: detailed_info of one channel
+        '''
+        channels_basic_info = self.channels_basic_info()
+        channels_detailed_info = {}
+        for channel in channels_basic_info:
+            channels_detailed_info[channel["cid"]] = self._requests_get(
+                'channels/{}'.format(channel["cid"]))
+        return channels_detailed_info
+
+    def channels_to_quarantine(self):
+        '''
+        Method is using channels_detailed_information for getting seconds_empty
+        and channel_topic values in order to specify if there should be put
+        status of quarantine on channels that meet certain condition.
+        There is no input taken from outside of class.
+        Output: Method is returning short information that it found channels
+        that needed attention.
+        '''
+        channels_detailed_info = self.channels_detailed_information()
+        for channel_id in channels_detailed_info:
+            channel = channels_detailed_info[channel_id]
+            if int(channel["seconds_empty"]) > self.qtime:
+                payload = {'channel_topic': 'quarantine'}
+                self._requests_post('channels/{}/topic'.format(channel_id), payload)
+        return "quarantined {}".format(payload)
```

Comment:
is this make sense to return a string? Isn't it part of a log?

Reply:
fixed, changed to log